sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosPagedIterableTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ *
+ */
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.InternalObjectNode;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.rx.TestSuiteBase;
+import com.azure.cosmos.util.CosmosPagedIterable;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Factory;
+import org.testng.annotations.Test;
+
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CosmosPagedIterableTest extends TestSuiteBase {
+
+    private static final int NUM_OF_ITEMS = 10;
+
+    private CosmosClient cosmosClient;
+    private CosmosContainer cosmosContainer;
+
+    @Factory(dataProvider = "clientBuilders")
+    public CosmosPagedIterableTest(CosmosClientBuilder clientBuilder) {
+        super(clientBuilder);
+    }
+
+    @BeforeClass(groups = { "simple" }, timeOut = SETUP_TIMEOUT)
+    public void before_CosmosPagedIterableTest() {
+        assertThat(this.cosmosClient).isNull();
+        this.cosmosClient = getClientBuilder().buildClient();
+        CosmosAsyncContainer asyncContainer = getSharedMultiPartitionCosmosContainer(this.cosmosClient.asyncClient());
+        cosmosContainer =
+            cosmosClient.getDatabase(asyncContainer.getDatabase().getId()).getContainer(asyncContainer.getId());
+        createItems(NUM_OF_ITEMS);
+    }
+
+    @AfterClass(groups = { "simple" }, timeOut = SHUTDOWN_TIMEOUT, alwaysRun = true)
+    public void afterClass() {
+        assertThat(this.cosmosClient).isNotNull();
+        this.cosmosClient.close();
+    }
+
+    @Test(groups = { "simple" }, timeOut = TIMEOUT)
+    public void readAllItemsByPageWithCosmosPagedIterableHandler() throws Exception {
+        CosmosQueryRequestOptions cosmosQueryRequestOptions = new CosmosQueryRequestOptions();
+
+        CosmosPagedIterable<InternalObjectNode> cosmosPagedIterable =
```

Comment:
ditto.
for the new test please don't use `InternalObjectNode` this will be used by the end user.

Please rely on ObjectNode or a simple POJO.